api/v1alpha1/serviceintentions_types_test.go

```diff
@@ -978,33 +925,37 @@ func TestServiceIntentions_ValidateNamespaces(t *testing.T) {
 					},
 				},
 			},
-			`serviceintentions.consul.hashicorp.com "does-not-matter" is invalid: [spec.sources[0].namespace: Invalid value: "bar": consul namespaces must be enabled to set source.namespace, spec.sources[1].namespace: Invalid value: "baz": consul namespaces must be enabled to set source.namespace, spec.sources[2].namespace: Invalid value: "baz": consul namespaces must be enabled to set source.namespace]`,
+			namespacesEnabled: false,
+			expectedErrMsg: []string{
+				`namespace: Invalid value: "namespaceA": consul namespaces must be enabled to set source.namespace`,
+				`namespace: Invalid value: "namespaceB": consul namespaces must be enabled to set source.namespace`,
+				`namespace: Invalid value: "namespaceC": consul namespaces must be enabled to set source.namespace`,
```

Comment:
Note that when there are multiple errors I did not assert on every character in the error message. Since the order is not deterministic, we don't know if `spec.sources[0].namespace` is going to refer to `namespaceA` or `namespaceB` or `namespaceC`, so I omitted that part of the assertion. From the current assertion, we will  know that all 3 of these strings are in the error message, so we are asserting on multiple errors, but not every character.

Reply:
Ah thank you for this catch! Makes sense.